build-system/tasks/build-bento.js

```diff
@@ -0,0 +1,271 @@
+const argv = require('minimist')(process.argv.slice(2));
+const debounce = require('../common/debounce');
+const {
+  bentoBundles,
+  jsBundles,
+  verifyBentoBundles,
+} = require('../compile/bundles.config');
+const {
+  INABOX_EXTENSION_SET,
+  buildBinaries,
+  buildExtensionCss,
+  buildExtensionJs,
+  buildNpmBinaries,
+  buildNpmCss,
+  declareExtension,
+  dedupe,
+  getBentoBuildFilename,
+  getExtensionsFromArg,
+} = require('./extension-helpers');
+const {analyticsVendorConfigs} = require('./analytics-vendor-configs');
+const {doBuildJs, endBuildStep, watchDebounceDelay} = require('./helpers');
+const {existsSync, mkdirSync} = require('fs');
+const {getBentoName} = require('./bento-helpers');
+const {log} = require('../common/logging');
+const {red} = require('kleur/colors');
+const {watch} = require('chokidar');
+
+// All declared components.
+const components = {};
+
+// All components to build.
+let componentsToBuild = null;
+
+/**
+ * Initializes all components from build-system/compile/bundles.config.bento.json
+ * if not already done and populates the given components object.
+ * @param {?Object} componentsObject
+ * @param {boolean=} includeLatest
+ */
+function maybeInitializeComponents(
+  componentsObject = components,
+  includeLatest = false
+) {
+  if (Object.keys(componentsObject).length === 0) {
+    verifyBentoBundles();
+    bentoBundles.forEach((c) => {
+      declareExtension(
+        c.name,
+        c.version,
+        c.latestVersion,
+        c.options,
+        componentsObject,
+        includeLatest
+      );
+    });
+  }
+}
+
+/**
+ * Process the command line arguments --nocomponents, --components, and
+ * --components_from and return a list of the referenced components.
+ *
+ * @param {boolean=} preBuild
+ * @return {!Array<string>}
+ */
+function getComponentsToBuild(preBuild = false) {
+  componentsToBuild = [];
+  if (argv.extensions) {
+    if (typeof argv.extensions !== 'string') {
+      log(red('ERROR:'), 'Missing list of components.');
+      process.exit(1);
+    } else if (argv.extensions === 'inabox') {
+      argv.extensions = INABOX_EXTENSION_SET.join(',');
+    }
+    const explicitComponents = argv.extensions.replace(/\s/g, '').split(',');
+    componentsToBuild = dedupe(componentsToBuild.concat(explicitComponents));
+  }
+  if (argv.extensions_from) {
+    const componentsFrom = getExtensionsFromArg(argv.extensions_from);
+    componentsToBuild = dedupe(componentsToBuild.concat(componentsFrom));
+  }
+  if (
+    !preBuild &&
+    !argv.nocomponents &&
+    !argv.extensions &&
+    !argv.extensions_from &&
+    !argv.core_runtime_only
+  ) {
+    const allComponents = [];
+    for (const component in components) {
+      allComponents.push(components[component].name);
+    }
```

Comment:
For your consideration:
```suggestion
    const allComponents = Object.values(components).map(c => c.name);
```